app/components/add-request.js

```diff
@@ -1,43 +1,61 @@
 import Ember from "ember";
 const { getOwner } = Ember;
-import AjaxPromise from 'stock/utils/ajax-promise';
+import AjaxPromise from "stock/utils/ajax-promise";
 
 export default Ember.Component.extend({
+  layoutName: null,
+  isGCRequest: null,
+
   store: Ember.inject.service(),
+  messageBox: Ember.inject.service(),
+  i18n: Ember.inject.service(),
   request: null,
   num: null,
   order: null,
 
-  packageTypeName: Ember.computed('request.packageType.name', function(){
-    return this.get('request.code.name');
+  packageTypeName: Ember.computed("request.code.name", function() {
+    return this.get("request.code.name");
+  }),
+
+  onInit: Ember.on("init", function() {
+    if (this.get("isGCRequest")) {
+      this.set("layoutName", "components/appointment-add-request");
+    } else {
+      this.set("layoutName", "components/order-add-request");
+    }
   }),
 
   actions: {
     deleteRequest(reqId) {
       var i18n = this.get("i18n");
       this.get("messageBox").custom(
-        i18n.t("order.request.remove_req").string + this.get("model.code"),
-        i18n.t("order.request.remove").string, () => this.send("removeRequest", reqId),
+        i18n.t("order.request.remove_req").string + this.get("order.code"),
```

Comment:
just a thought, I think `i18n` supports string interpolation (not sure though).
It'd be worth testing if we can do something like:
```
I18n.translations = {
  en: {
    hello: 'hello {{name}}',
    }
  }
}

// hbs
{{ t 'hello' name=myName }}

```

Reply:
@patrixr , yes it does support it.

CC: @abulsayyad123

Reply:
Yes it is possible, and it has implemented where we need to pass attributes to translations. In this case, message-box is used at many places which require to add changes at many place. I think we can take away this point for future reference.

Reply:
It's not change in structure of messageBox.